backend/cfg/eliminate_fallthrough_blocks.ml

```diff
@@ -71,7 +71,13 @@ let rec disconnect_fallthrough_blocks cfg_with_layout =
     if !C.verbose then
       Printf.printf "%s: disconnected fallthrough blocks: %d\n" cfg.fun_name
         len;
-    disconnect_fallthrough_blocks cfg_with_layout )
+    disconnect_fallthrough_blocks cfg_with_layout;
+    begin match cfg.fun_tailrec_entry_point_label with
+    | None -> ()
+    | Some label ->
+      if List.exists (Label.equal label) found then
+        Cfg.set_fun_tailrec_entry_point_label cfg None
+    end)
```

Comment:
Same as above, I think this check should be done at the end of the entire transformation.

Unlike the above case, this seem correct for a subtle reason: if tailrec block was a fallthrough and got eliminated,  then either (a) there are Tailcall Self, and then the `label` here will be the new `fun_tailrec_entry_point_label` which is the successor of the original one, so won't be in `found` or (b) there are no Tailcall Self, and the label is not used and can be safely set to None.